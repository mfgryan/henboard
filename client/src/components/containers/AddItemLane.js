// redux dep
import { connect } from "react-redux";
import { addItem } from "../../actions/items";
import { toggleAddItem, changeValue } from "../../actions/lanes";
import projects from "../../models/projects.js";
import sprints from "../../models/sprints.js";
import lanes from "../../models/lanes.js";

// component dep
import AddItem from "../AddItem";

const mapStateToProps = (state, ownProps) => {
    let project = projects.getCurrentProject(state);
    let sprint = sprints.getCurrentSprint(state);
    let lane = lanes.getLane(state, project.project, ownProps.column);
    return {
        title: ownProps.column,
        value: lane.value,
        showAddItem: lane.addItem,
        item: {
            project: project.project,
            column: ownProps.column,
            week: sprint.week,
            name: lane.value
        }
    };
};

const mapDispatchToProps = (dispatch, ownProps) => {
    return {
        toggleAddItem: title => {
            dispatch(
                toggleAddItem({
                    project: ownProps.project,
                    column: title
                })
            );
        },
        changeValue: (event, title) => {
            dispatch(
                changeValue({
                    project: ownProps.project,
                    column: title,
                    value: event.target.value
                })
            );
        },
        addItem: item => {
            dispatch(
                addItem({
                    project: item.project,
                    week: item.week,
                    column: item.column,
                    name: item.name
                })
            );
            //after adding an item clear the input field
            dispatch(
                changeValue({
                    project: item.project,
                    column: item.column,
                    value: ""
                })
            );
        }
    };
};

const AddItemLane = connect(mapStateToProps, mapDispatchToProps)(AddItem);

export default AddItemLane;
