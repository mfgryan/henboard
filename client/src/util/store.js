// redux dep
import { createStore, applyMiddleware } from "redux";
import { logger, validate } from "../util/middleware.js";
import data from "../util/data.js";
import rootReducer from "../reducers/index.js";

const store = createStore(rootReducer,applyMiddleware(logger,validate));

export { store }
