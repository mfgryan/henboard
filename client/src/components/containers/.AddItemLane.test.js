import React from "react";
import AddItemLane from "./AddItemLane";
import renderer from "react-test-renderer";

// redux dep
import { Provider } from "react-redux";
import store from "../../util/store.js";

// router dep
import { BrowserRouter as Router } from "react-router-dom";

it("AddItemLane renders correctly", () => {
    const tree = renderer
        .create(
            <Router>
                <Provider store={store}>
                    <AddItemLane project="henboard" column="Todo" />
                </Provider>
            </Router>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
