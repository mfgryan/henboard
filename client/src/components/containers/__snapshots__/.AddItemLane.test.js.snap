exports[`test AddItemLane renders correctly 1`] = `
<div
  className="AddItem">
  <h2
    onClick={[Function]}>
    Todo
  </h2>
  <div
    className="AddItemInput"
    style={
      Object {
        "visibility": "hidden",
      }
    }>
    <div
      className="form-group">
      <span
        className="input-group input-group-sm">
        <input
          className="form-control"
          id={undefined}
          onChange={[Function]}
          onKeyPress={[Function]}
          type="text"
          value="" />
        <span
          className="input-group-btn"
          type="text">
          <button
            className="btn btn-default"
            disabled={false}
            onClick={[Function]}
            type="button">
            Add
          </button>
        </span>
      </span>
    </div>
  </div>
</div>
`;
